packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -226,10 +228,18 @@ Future<void> writeBundle(
     bundleDir.deleteSync(recursive: true);
   bundleDir.createSync(recursive: true);
 
+  // On systems with a limited number of file descriptors, limit number of
+  // open files.
+  final Pool pool = Pool(platform.isMacOS ? 30 : 1000);
```

Comment:
How about something like 64 everywhere. We'll hit the same problem again on some other platform with a small limit, and I'd be surprised if there was any performance impact on any platform.